src/main/java/org/opentripplanner/routing/algorithm/filterchain/filters/system/FlexSearchWindowFilter.java

```diff
@@ -0,0 +1,67 @@
+package org.opentripplanner.routing.algorithm.filterchain.filters.system;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.function.Predicate;
+import org.opentripplanner.model.plan.Itinerary;
+import org.opentripplanner.model.plan.SortOrder;
+import org.opentripplanner.routing.algorithm.filterchain.framework.spi.RemoveItineraryFlagger;
+
+/**
+ * The flex router doesn't use the transit router's  search-window, but nevertheless using it
+ * for filtering is useful when combining flex with transit.
+ * <p>
+ * The flex router also searches the previous day (arrive by) or the next one (depart after).
+ * If you didn't filter the flex results by something you could get yesterday's or tomorrow's
+ * trips where you would not expect it.
+ */
+public class FlexSearchWindowFilter implements RemoveItineraryFlagger {
+
+  public static final String TAG = "flex-outside-search-window";
+
+  private final Instant earliestDepartureTime;
+  private final Instant latestArrivalTime;
+  private final SortOrder sortOrder;
+
+  public FlexSearchWindowFilter(
+    Instant earliestDepartureTime,
+    Duration searchWindow,
+    SortOrder sortOrder
+  ) {
+    this.earliestDepartureTime = earliestDepartureTime;
+    this.latestArrivalTime = earliestDepartureTime.plus(searchWindow);
+    this.sortOrder = sortOrder;
+  }
+
+  @Override
+  public String name() {
+    return TAG;
+  }
+
+  @Override
+  public Predicate<Itinerary> shouldBeFlaggedForRemoval() {
+    return it -> {
+      if (it.isDirectFlex()) {
+        return switch (sortOrder) {
+          case STREET_AND_DEPARTURE_TIME -> {
+            // arive by
```

Comment:
```suggestion
```
The reason why we use the sortOrder, and not arriveBy is that the sort order is not the same as the arriveBy flag in all cases. The paging changes this, but the logic here should use sort order, and anyone reading this code or changing it should read the JavaDoc on sort order whitch I agree does not spell it out, but at least hints to it.